rpcserver.go

```diff
@@ -637,6 +637,73 @@ func createVinList(mtx *wire.MsgTx) []btcjson.Vin {
 	return vinList
 }
 
+// createVinList returns a slice of JSON objects for the inputs of the passed
+// transaction.
+func createVinListPrevOut(s *rpcServer, mtx *wire.MsgTx, chainParams *chaincfg.Params, vinExtra int) []btcjson.VinPrevOut {
+	vinList := make([]btcjson.VinPrevOut, len(mtx.TxIn))
+
+	// we don't call fetchInputTransactions() here because it may not be
+	// necessary so we defer until it is.  ( lazy load )
+	var txStore *blockchain.TxStore
+
+	for i, v := range mtx.TxIn {
+		if blockchain.IsCoinBaseTx(mtx) {
+			vinList[i].Coinbase = hex.EncodeToString(v.SignatureScript)
+		} else {
+			vinList[i].Txid = v.PreviousOutPoint.Hash.String()
+			vinList[i].Vout = v.PreviousOutPoint.Index
+
+			// The disassembled string will contain [error] inline
+			// if the script doesn't fully parse, so ignore the
+			// error here.
+			disbuf, _ := txscript.DisasmString(v.SignatureScript)
+			vinList[i].ScriptSig = new(btcjson.ScriptSig)
+			vinList[i].ScriptSig.Asm = disbuf
+			vinList[i].ScriptSig.Hex = hex.EncodeToString(v.SignatureScript)
+
+			// If vinExtra flag is set then we grab extra data from the
+			// previous transaction output.
+			if vinExtra == 1 {
+
+				// we only need to do this the first time because fetchInputTransactions
+				// gives us all the input transactions in one go.
+				if txStore == nil {
+					tx := btcutil.NewTx(mtx)
+					txStoreNew, err := s.server.txMemPool.fetchInputTransactions(tx, true)
+					if err == nil {
+						txStore = &txStoreNew
+					}
+				}
+				if txStore != nil && len(*txStore) != 0 {
+
+					vinList[i].PrevOut = new(btcjson.PrevOut)
+
+					txData := (*txStore)[v.PreviousOutPoint.Hash]
```

Comment:
This isn't safe since it's assuming the utxo is there.  If it's not, it would result in a panic when dereferencing the nil `txData`. See my previous comment above how it is checked for nil.